### PR TITLE
Add LoRA conversion from MS to PT

### PR DESCRIPTION
**Pull request by SamitHuang**

### CODE REVIEW

File: examples/stable_diffusion_xl/tools/lora_conversion/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+The converted checkpoint will be saved in the same folder of {path to ms_ckpt}.
+
+## Run inference in diffusers
+
+Specify the model path with the converted checkpoint in diffusers
+
+```
+python infer_lora.py --model_path {path to converted ckpt}
+```
```
Comment (vigo999) [inline examples/stable_diffusion_xl/tools/lora_conversion/README.md:18]: 能否增加一些ms 和 pt的生成图片

**Review by wtomin — APPROVED** [review]

File: examples/stable_diffusion_xl/tools/lora_conversion/README.md (hunk 1 above)
Comment (SamitHuang) [reply examples/stable_diffusion_xl/tools/lora_conversion/README.md:19]: Done

File: examples/stable_diffusion_xl/demo/sampling_without_streamlit.py [hunk 2]
```diff
@@ -82,15 +92,30 @@ def get_parser_sample():
...
-    args, model, version_dict, is_legacy=False, return_latents=False, filter=None, stage2strength=None, amp_level="O0"
+    args,
+    model,
+    version_dict,
+    is_legacy=False,
+    return_latents=False,
+    filter=None,
+    stage2strength=None,
+    amp_level="O0",
+    save_path="./",
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/demo/sampling_without_streamlit.py:103]: `save_path` has already been defined in args

File: examples/stable_diffusion_xl/demo/sampling_without_streamlit.py [hunk 3]
```diff
@@ -297,6 +334,7 @@ def sample(args):
             filter=filter,
             stage2strength=stage2strength,
             amp_level=args.ms_amp_level,
+            save_path=save_path,
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/demo/sampling_without_streamlit.py:337]: ditto

File: examples/stable_diffusion_xl/demo/sampling_without_streamlit.py [hunk 4]
```diff
@@ -311,10 +349,11 @@ def sample(args):
...
         raise ValueError(f"Unknown task {task}")
 
-    out = out if isinstance(out, (tuple, list)) else [out, None]
-    (samples, samples_z) = out
+    if task != "txt2img":
+        out = out if isinstance(out, (tuple, list)) else [out, None]
+        (samples, samples_z) = out
 
-    perform_save_locally(save_path, samples)
+        perform_save_locally(save_path, samples)
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/demo/sampling_without_streamlit.py:356]: Is it more consistent to move it to `run_img2img`

File: examples/stable_diffusion_xl/demo/sampling_without_streamlit.py [hunk 5]
```diff
@@ -337,5 +376,13 @@ def sample(args):
...
     args, _ = parser.parse_known_args()
-    ms.context.set_context(mode=args.ms_mode, device_target=args.device_target)
+    if args.precision_keep_origin_dtype:
+        ms.context.set_context(
+            mode=args.ms_mode,
+            device_target=args.device_target,
+            ascend_config=dict(precision_mode="must_keep_origin_dtype"),
+        )  # NOTE: Needed for aligning with diffusers
+    else:
+        ms.context.set_context(mode=args.ms_mode, device_target=args.device_target)
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/demo/sampling_without_streamlit.py:386]: 1. is this necessary？

2. Should the comment section be removed as the code base is refer to official implementation

3. modify to:
```
ms.context.set_context(mode=args.ms_mode, device_target=args.device_target)
if args.precision_keep_origin_dtype:
    ms.context.set_context(ascend_config=dict(precision_mode="must_keep_origin_dtype"))
```

File: examples/stable_diffusion_xl/tools/lora_conversion/README.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+      --save_path outputs/lora_pokemon_fix_noise \
+      --weight $base_ckpt_path,$ckpt_path \
+      --prompt "a black and white photo of a butterfly" \
+      --init_latent_path $init_latent_path \
+      --device_target Ascend \
+      --discretization "DiffusersDDPMDiscretization" \
+      --precision_keep_origin_dtype True \
+    ```
+
+    Note that if you use diffusers for comparison, you should set discretization and precision mode as above to minimize the computational difference.
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/tools/lora_conversion/README.md:79]: 是否 `diffuser` -> `official/diffuser `or `official` 更好

File: examples/stable_diffusion_xl/scripts/run_train_lora.sh [hunk 7]
```diff
@@ -0,0 +1,25 @@
+export DEVICE_ID=$1
+export MS_PYNATIVE_GE=1
+# export MS_ASCEND_CHECK_OVERFLOW_MODE="INFNAN_MODE"
+# export MS_ASCEND_CHECK_OVERFLOW_MODE=1 # "INFNAN_MODE"
+export PYTHONPATH=$(pwd):$PYTHONPATH
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/scripts/run_train_lora.sh:5]: The training process does not require these two environmental variables

Comment (SamitHuang) [reply examples/stable_diffusion_xl/scripts/run_train_lora.sh]: fixed

File: examples/stable_diffusion_xl/demo/sampling_without_streamlit.py (hunk 5 above)
Comment (SamitHuang) [reply examples/stable_diffusion_xl/demo/sampling_without_streamlit.py]: 1. yes if want to minimize the difference btw diffusers and ms inference.
2. removed
3. updated

File: examples/stable_diffusion_xl/tools/lora_conversion/README.md (hunk 6 above)
Comment (SamitHuang) [reply examples/stable_diffusion_xl/tools/lora_conversion/README.md:79]: i think it's a common term for users

**Pull request merged**